flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -292,6 +294,14 @@ protected void resubmitJob(FlinkResourceContext<CR> ctx, boolean requireHaMetada
         restoreJob(ctx, specToRecover, ctx.getObserveConfig(), requireHaMetadata);
     }
 
+    protected void cancelJob(FlinkResourceContext<CR> ctx) throws Exception {
```

Comment:
Do we need to introduce a new method? I would prefer to keep the existing ones because the third method with the same name is getting a bit confusing